From 9cbb505ff6a9cb781ca95ea1af612297e5672837 Mon Sep 17 00:00:00 2001
Message-Id: <9cbb505ff6a9cb781ca95ea1af612297e5672837.1650636481.git.keller@fs-net.de>
In-Reply-To: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
References: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
From: bings <bings@codeaurora.org>
Date: Mon, 13 Aug 2018 16:56:50 +0800
Subject: [PATCH 12/56] qcacld-2.0: Always register frames for supplicant in
 hdd_wlan_startup

When there is P2P GO interface, con_mode is VOS_STA_SAP_MODE. At this time
if surprise remove happens, frames for supplicant will not be registered in
driver probe.
Always register frames for supplicant in hdd_wlan_startup.

Change-Id: Ifbf346f44b42e63dde1095fb7d3ac0a685105d28
CRs-Fixed: 2293888
---
 CORE/HDD/src/wlan_hdd_main.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/CORE/HDD/src/wlan_hdd_main.c b/CORE/HDD/src/wlan_hdd_main.c
index 3738635..b19dd1a 100644
--- a/CORE/HDD/src/wlan_hdd_main.c
+++ b/CORE/HDD/src/wlan_hdd_main.c
@@ -17819,13 +17819,7 @@ int hdd_wlan_startup(struct device *dev, v_VOID_t *hif_sc)
    }
 
    hdd_register_mcast_bcast_filter(pHddCtx);
-   if (VOS_STA_SAP_MODE != hdd_get_conparam())
-   {
-      /* Action frame registered in one adapter which will
-       * applicable to all interfaces
-       */
-      wlan_hdd_cfg80211_register_frames(pAdapter);
-   }
+   wlan_hdd_cfg80211_register_frames(pAdapter);
 
    mutex_init(&pHddCtx->sap_lock);
 
-- 
2.14.5

